source/postprocess/sea_level.cc

```diff
@@ -0,0 +1,584 @@
+        {
+          std::ofstream file (filename.c_str());
+
+          file << "# "
+               << "R phi theta"
+               << " nonuniform_sea_level_change"
+               << " sea_level_offset"
+               << " geoid_displacement"
+               << " topography"
+               << " ice_height"
+               << " ocean_mask" << std::endl;
+
+          // First write out the data we have created locally.
+          file << output_file.str();
+
+          std::string tmp;
+          tmp.resize (max_data_length, '\0');
+
+          // Then loop through all of the other processors and collect
+          // data, then write it to the file.
+          for (unsigned int p=1; p<Utilities::MPI::n_mpi_processes(this->get_mpi_communicator()); ++p)
+            {
+              MPI_Status status;
+              // Get the data. Note that MPI says that an MPI_Recv may receive
+              // less data than the length specified here. Since we have already
+              // determined the maximal message length, we use this feature here
+              // rather than trying to find out the exact message length with
+              // a call to MPI_Probe.
+              const int ierr = MPI_Recv (&tmp[0], max_data_length, MPI_CHAR, p, mpi_tag,
+                                         this->get_mpi_communicator(), &status);
+              AssertThrowMPI(ierr);
+
+              // Output the string. Note that 'tmp' has length max_data_length,
+              // but we only wrote a certain piece of it in the MPI_Recv, ended
+              // by a \0 character. Write only this part by outputting it as a
+              // C string object, rather than as a std::string.
+              file << tmp.c_str();
+            }
+        }
+      else
+        // On other processors, send the data to processor zero. include the \0
+        // character at the end of the string.
+        {
+          output_file << "\0";
+          const int ierr = MPI_Send (&output_file.str()[0], output_file.str().size()+1, MPI_CHAR, 0, mpi_tag,
+                                     this->get_mpi_communicator());
+          AssertThrowMPI(ierr);
+        }
+
+      // if output_interval is positive, then update the last supposed output time
+      if (output_interval > 0)
+        {
+          // We need to find the last time output was supposed to be written.
+          // This is the last_output_time plus the largest positive multiple
+          // of output_intervals that passed since then. We need to handle the
+          // edge case where last_output_time+output_interval==current_time,
+          // we did an output and std::floor sadly rounds to zero. This is done
+          // by forcing std::floor to round 1.0-eps to 1.0.
+          const double magic = 1.0+2.0*std::numeric_limits<double>::epsilon();
+          last_output_time = last_output_time + std::floor((this->get_time()-last_output_time)/output_interval*magic) * output_interval/magic;
+        }
```

Comment:
I made a comment above about some other plugin that had broken this piece of functionality into a separate function. Feel free to do as suggested there, or ignore my suggestion.